dbms/src/Interpreters/SettingsCommon.h

```diff
@@ -772,15 +772,19 @@ struct SettingCompressionMethod
             return CompressionMethod::LZ4HC;
         if (lower_str == "zstd")
             return CompressionMethod::ZSTD;
+#ifdef ENABLE_QPL_COMPRESSION
```

Comment:
This is one weird thing of what I mentioned in my other comment. If user specifies "qpl" in config file but uses a non-qpl binary, the user sees a error message `Unknown compression method: 'qpl', must be one of 'lz4', 'lz4hc', 'zstd', 'qpl'` - pretty confusing.

Reply:
Hope this helps explaining what I meant by we either ... or ...